### PR TITLE
[IMPORTANT] UserRuleIntent field in Teacher

### PR DESCRIPTION
**Pull request by epawlowska**
- changes order of arguments in UserRuleIntent constructor (!)
- adds UserRuleIntent field in Teacher

### CODE REVIEW

File: whylog/teacher/__init__.py [hunk 1]
```diff
@@ -19,8 +20,7 @@ class Teacher(object):
 
     def __init__(self, id_to_line_dict, effect_id, config, pattern_assistant):
         self._lines = id_to_line_dict.copy()
-        self.effect_id = effect_id
-        self.constraints = set()
+        self.rule = UserRuleIntent(effect_id, parsers={}, constraints=[])
```
Comment (ppolewicz) [inline whylog/teacher/__init__.py]: `UserRuleIntent` should be instantiated just before being passed to Config. It is a DAO, which cannot comprehend things like history or relationships between items, or methods on them. The only reason this class exists is to pass it between Teacher and Config.

Comment (epawlowska) [inline whylog/teacher/__init__.py]: I think it will be ok to use UserRuleIntent also as an container for information about rule.

Comment (ppolewicz) [inline whylog/teacher/__init__.py]: Maybe it is ok. Please rename the field and move default values to constructor.

File: whylog/teacher/__init__.py [hunk 2]
```diff
@@ -19,8 +20,7 @@ class Teacher(object):
 
     def __init__(self, id_to_line_dict, effect_id, config, pattern_assistant):
         self._lines = id_to_line_dict.copy()
-        self.effect_id = effect_id
-        self.constraints = set()
+        self.rule_intent = UserRuleIntent(effect_id, parsers=None, constraints=None)
```
Comment (ppolewicz) [inline whylog/teacher/__init__.py]: `self.rule_intent = UserRuleIntent(effect_id)`

File: whylog/teacher/user_intent.py [hunk 3]
```diff
@@ -14,7 +14,7 @@ def __init__(self, constr_type, groups, params=None):
 
 
 class UserRuleIntent(object):
-    def __init__(self, parsers, effect_id, constraints):
-        self.parsers = parsers
+    def __init__(self, effect_id, parsers, constraints):
```
Comment (ppolewicz) [inline whylog/teacher/user_intent.py]:
``` python
def __init__(self, effect_id, parsers=None, constraints=None):
    self.constraints = constraints or {}
    self.parsers = parsers or {}
```

**Pull request merged**